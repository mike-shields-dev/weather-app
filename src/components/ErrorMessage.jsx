import React from "react";
import PropTypes from "prop-types";
import styles from "../styles/ErrorMessage.module.css";

const ErrorMessage = ({ errorMessage, handleClick }) => {
  return (
    <div className={styles["error-message"]}>
      <span className={styles["error-message__text"]}>{errorMessage}</span>
      <button
        className={styles["error-message__button"]}
        onClick={handleClick}
        type="button"
      >
        OK
      </button>
    </div>
  );
};

ErrorMessage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default ErrorMessage;
