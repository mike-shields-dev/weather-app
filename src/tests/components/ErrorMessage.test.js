import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ErrorMessage from "../../components/ErrorMessage";

const validProps = {
  errorMessage: "City not found",
  handleClick: jest.fn(),
};

describe("ErrorMessage", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<ErrorMessage {...validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders an error message", () => {
    const { getByText } = render(<ErrorMessage {...validProps} />);

    expect(getByText(validProps.errorMessage)).toBeInTheDocument();
  });

  it("renders an 'OK' button", () => {
    const { getByText } = render(<ErrorMessage {...validProps} />);

    expect(getByText("OK")).toBeInTheDocument();
  });

  it("OK button calls handleClick callback", () => {
    const { getByText } = render(<ErrorMessage {...validProps} />);
    const okButton = getByText("OK");

    fireEvent.click(okButton);

    expect(validProps.handleClick).toHaveBeenCalled();
  });
});
