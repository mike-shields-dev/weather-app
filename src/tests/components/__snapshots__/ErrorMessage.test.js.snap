// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="error-message"
  >
    <span
      class="error-message__text"
    >
      City not found
    </span>
    <button
      class="error-message__button"
      type="button"
    >
      OK
    </button>
  </div>
</DocumentFragment>
`;
